theano/gpuarray/extra_ops.py

```diff
@@ -430,8 +430,8 @@ def c_support_code_struct(self, node, nodename):
                         size_t dimBlock[3] = {1, 1, 1};
                         size_t tmp0 = shape[axis]-2;
                         size_t tmp1 = shape[axis]-1;
-                        void* kernel_params[] = {(void*) input->ga.data,
-                                                 (void*) output->ga.data,
+                        void* kernel_params[] = {(void*) ((char *)(input->ga.data) + input->ga.offset),
```

Comment:
I think `input->ga.data` is the address of the beginning of the memory segment, not the address of the first element, and we need to add the offset to get to the first element. Am I missing something?